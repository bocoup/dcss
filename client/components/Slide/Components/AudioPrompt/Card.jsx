import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="microphone" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
