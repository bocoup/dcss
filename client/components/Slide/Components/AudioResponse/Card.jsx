import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const AudioResponseCard = () => (
  <Popup
    content="An audio recorder for users to record responses."
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="microphone" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(AudioResponseCard);
