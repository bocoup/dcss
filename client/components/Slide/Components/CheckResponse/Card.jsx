import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';

const TextResponseCard = () => (
    <Popup
        content="A checkbox response"
        header="Checkbox Response"
        trigger={
            <React.Fragment>
                <Icon
                    name="check square outline"
                    aria-label="Checkbox Response"
                />
                Checkbox Response
            </React.Fragment>
        }
    />
);

export default React.memo(TextResponseCard);
