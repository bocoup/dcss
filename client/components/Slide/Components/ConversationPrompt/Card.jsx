import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="conversation" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
