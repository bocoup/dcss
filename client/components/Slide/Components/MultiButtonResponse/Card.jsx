import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const MultiButtonResponseCard = () => (
  <Popup
    content="A multiple button response"
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="hand pointer outline" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(MultiButtonResponseCard);
