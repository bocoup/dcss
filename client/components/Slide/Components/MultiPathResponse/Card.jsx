import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="fork" rotated="clockwise" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
