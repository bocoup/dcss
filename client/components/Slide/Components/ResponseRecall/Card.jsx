import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const ResponseRecallCard = () => (
  <Popup
    content="Embed a previously entered participant response"
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="comment alternate outline" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(ResponseRecallCard);
