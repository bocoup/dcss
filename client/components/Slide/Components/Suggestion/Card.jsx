import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="info" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
