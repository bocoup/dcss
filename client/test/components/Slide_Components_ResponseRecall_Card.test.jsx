import {
  mounter,
  reduxer,
  snapshot,
  state
} from '../bootstrap';
import {
  unmountComponentAtNode
} from 'react-dom';
import {
  cleanup,
  fireEvent,
  render
} from '@testing-library/react';
import Card from '../../components/Slide/Components/ResponseRecall/Card.jsx';

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetch.mockImplementation(() => {
    return Promise.resolve({
      status: 200,
      json() {
        return Promise.resolve({});
      },
    });
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const sharedProps = {
  history: {
    push() {},
  },
};

test('Card', () => {
  expect(Card).toBeDefined();
});

test('Snapshot 1', () => {
  const props = {
    ...sharedProps,
  };
  const mounted = mounter(reduxer(Card, props, state))();
  expect(snapshot(mounted)).toMatchSnapshot();
});

/*{INJECTION}*/
