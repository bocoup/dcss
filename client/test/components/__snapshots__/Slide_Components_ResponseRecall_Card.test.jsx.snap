// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1 1 1`] = `"<i aria-label=\\"Embed Previous Response\\" class=\\"comment alternate outline icon\\"></i>Embed Previous Response"`;

exports[`Snapshot 1 1 2`] = `"<i aria-label=\\"Embed Previous Response\\" class=\\"comment alternate outline icon\\"></i>Embed Previous Response"`;

exports[`Snapshot 1 1 3`] = `"<i aria-label=\\"Embed Previous Response\\" class=\\"comment alternate outline icon\\"></i>Embed Previous Response"`;

exports[`Snapshot 1 2 1`] = `"<i aria-label=\\"Embed Previous Response\\" class=\\"comment alternate outline icon\\"></i>Embed Previous Response"`;
